ehsm_kms_service/function.js

```diff
@@ -601,33 +601,52 @@ const _query_api_key = async (DB, appid) => {
 }
 
 /**
- * Gen Hmac 
+ * Gen Hmac of sign_params safely, without transfering plaintext of apikey outside enclave
  * @param {object} DB //Database object
  * @param {string} appid // APP ID from client
  * @param {object} sign_params // A list object contain string params
  * @returns {object} (error, hmac) // One object contain "error" and "hmac" attributes
  */
 const gen_hmac = async (DB, appid, sign_params) => {

```

Comment:
not updated? using gen_hmac_safe?

Reply:
Currently the original implementation `gen_hmac()` function is removed, while `gen_hmac_safe()` function has been renamed to `gen_hmac()`.
Since the original implementation of `gen_hmac()` and current implementation (same as `gen_hmac_safe()`) shares the same logic and have same output for certain input, the rename (`gen_hmac_safe()` -> `gen_hmac()`) should be safe.